quickstarts/microsoft.network/nat-gateway-vnet/azuredeploy.json

```diff
@@ -123,24 +128,27 @@
         ],
         "enableDdosProtection": false,
         "enableVmProtection": false
-      }
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/natGateways', parameters('natGatewayName'))]"
+      ]
     },
     {
```

Comment:
It seems to me this subnet resource is completely redundant, as it's declared under the vnet, and should be removed.  Do you agree?